heaps/min_heap.py

```diff
@@ -15,39 +15,63 @@ class MinHeap:
     def __init__(self):
         self.store = []
 
+    # helpers
+    def parent_index(self, index):
+        return (index - 1) // 2
+
+    def left_child_index(self, index):
+        return (2 * index) + 1
+
+    def right_child_index(self, index):
+        return (2 * index) + 2
+
 
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(1)
         """
-        pass
+
+        if value is None:
+            value = key
+
+        new_node = HeapNode(key, value)
+        self.store.append(new_node)
+        self.heap_up(len(self.store) - 1) 
+
 
     def remove(self):
         """ This method removes and returns an element from the heap
             maintaining the heap structure
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(1)
```

Comment:
✨ Because of the recursive call stack of `heap_down` space complexity is O(log n)